### PR TITLE
Pull topshare from supabase backend

### PR DESCRIPTION
**Pull request by seren5240**
Currently there is a UI glitch when rendering the topshare (displays regular player info as connection is established to database which changes to include topshare once topshares are returned). Will attempt to eliminate next.

Edit: IMO changed to satisfactory behaviour by [135dc09](https://github.com/morgante/diploassist/pull/2/commits/135dc098bd1d28442f3102e6adb3ad3ebb2c30ab)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,4 +2,5 @@ backstabbr.js
 node_modules/*
 .idea/*
 package-lock.json
-built/
\ No newline at end of file
+built/
+package.json
```
Comment (morgante) [inline .gitignore]: nit: files should end with a newline

**Review by morgante — CHANGES_REQUESTED** [review]
Just one nit, otherwise LGTM! 🚀

File: .gitignore (hunk 1 above)
Comment (seren5240) [reply .gitignore]: TIL! Configured in VS Code :)

File: src/convoys.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    didConvoyFail: boolean
+): void {
+const convoy = new Convoy(
+    centerOfConvoyingUnit,
+    centerOfConvoyedUnit,
+    centerOfConvoyDestination,
+    didConvoyFail
+);
+
+convoy.drawConvoyAssistPath();
```
Comment (morgante) [inline src/convoys.ts]: These should be indented.

It would be a good idea to set up prettier on this repo: https://prettier.io/

**Review by morgante — CHANGES_REQUESTED** [review]

**Review by morgante — APPROVED** [review]

**Pull request merged**